Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp

```diff
@@ -104,11 +115,59 @@ static void wpeDisplayDispose(GObject* object)
     G_OBJECT_CLASS(wpe_display_parent_class)->dispose(object);
 }
 
+static void wpeDisplaySetProperty(GObject* object, guint propId, const GValue* value, GParamSpec* paramSpec)
+{
+    auto* display = WPE_DISPLAY(object);
+
+    switch (propId) {
+    case PROP_AVAILABLE_INPUT_TYPES:
+        if (display->priv->availableInputTypes != g_value_get_flags(value))  {
+            display->priv->availableInputTypes =  static_cast<WPEAvailableInputTypes>(g_value_get_flags(value));
+            g_object_notify_by_pspec(object, paramSpec);
+        }
```

Comment:
We should add a public setter and call it from here. Just document in the API docs of the setter that it should only be called by platform implementations